:books: Improve technical guide (#5881)
uxbox/uxbox: docs/technical-guide/configuration.md

PENPOT_SMTP_TLS: true
```

If you are not using SMTP configuration and want to log the emails in the console, you should use the following flag:

```bash
PENPOT_FLAGS: [...] enable-log-emails
```

## Redis

> The Redis configuration is very simple, just provide with a valid redis URI. Redis is used

Reviewer: ```suggestion
The Redis configuration is very simple, just provide a valid redis URI.
```

```bash
PENPOT_FLAGS: disable-registration enable-demo-users enable-demo-warning
```

**disable-registration** prevents any user from registering in the platform.
**enable-demo-users** creates users with a default expiration time of 7 days, and
once expired they are completely deleted with all the generated content.
From the registration page, there is a link with a `Create demo account` which creates one of these
users and logs in automatically.
**enable-demo-warning** is a modal in the registration and login page saying that the
environment is a testing one and the data may be wiped without notice.

Another way to work in a demo environment is allowing users to register but removing the
verification process:

```bash
PENPOT_FLAGS: disable-email-verification enable-demo-warning
```

## Backend

This section enumerates the backend only configuration variables.

### Secret key

The <code class="language-bash">PENPOT_SECRET_KEY</code> envvar serves a master key from which other keys
for subsystems (eg http sessions, or invitations) are derived.

If you don't use it, all created sessions and invitations will become invalid on container restart
or service restart.

> To use it, we recommend to use a trully randomly generated 512 bits base64 encoded string here.

Reviewer: ```suggestion
To use it, we recommend using a truly randomly generated 512 bits base64 encoded string here.
```

You can generate one with:

```bash
python3 -c "import secrets; print(secrets.token_urlsafe(64))"
```

And configure it:
```bash
# Backend
PENPOT_SECRET_KEY: my-super-secure-key
```

### Database

Penpot only supports PostgreSQL and we highly recommend >=13 version. If you are using official
docker images this is already solved for you.

Essential database configuration:

```bash
# Backend
PENPOT_DATABASE_USERNAME: penpot
PENPOT_DATABASE_PASSWORD: penpot
PENPOT_DATABASE_URI: postgresql://127.0.0.1/penpot
```

The username and password are optional. These settings should be compatible with the ones
in the postgres configuration:

```bash
# Postgres
POSTGRES_DATABASE: penpot
POSTGRES_USER: penpot
POSTGRES_PASSWORD: penpot
```

### Storage

> Storage refers to storage used for store the user uploaded assets.

Reviewer: ```suggestion
Storage refers to storing the user uploaded assets.
```

using environment variables, and the image automatically generates the <code class="language-bash">config.js</code> from
them.

> In case you have a custom setup you probably need setup the following environment

Reviewer: ```suggestion
In case you have a custom setup, you probably need to setup the following environment
```

These variables are used for generate correct nginx.conf file on container startup.

## Other flags

There are other flags that are useful for a more customized Penpot experience. This section has the list of the flags meant
for the user:

- <code class="language-bash">enable-cors</code>: Enables the default cors cofiguration that allows all domains
  (this configuration is designed only for dev purposes right now)
- <code class="language-bash">enable-backend-api-doc</code>: Enables the <code class="language-bash">/api/doc</code>
  endpoint that lists all rpc methods available on backend
- <code class="language-bash">disable-login-with-password</code>: allows disable password based login form
- <code class="language-bash">enable-prepl-server</code>: enables PREPL server, used by manage.py and other additional

>   tools for communicate internally with Penpot backend. Check the [CLI section][5] to get more detail.

Reviewer: ```suggestion
  tools to communicate internally with Penpot backend. Check the [CLI section][5] to get more detail.
```
